cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -290,9 +290,6 @@ class ARROW_EXPORT FileSystem : public std::enable_shared_from_this<FileSystem>
   /// Open an output stream for appending.
   ///
   /// If the target doesn't exist, a new empty file is created.
-  ARROW_DEPRECATED(
```

Comment:
I would add a note in the docstring that this cannot be supported by all filesystem kinds.